app/streams/publishing_api_consumer.rb

```diff
@@ -11,24 +11,56 @@ def process(message)
 private
 
   def process_message(message)
-    content_id = message.payload['content_id']
-    base_path = message.payload['base_path']
+    content_id = message.payload.fetch('content_id')
+    base_path = message.payload.fetch('base_path')
+    payload_version = message.payload.fetch('payload_version')
     locale = message.payload['locale'] || 'en'
 
     item = Dimensions::Item.by_natural_key(content_id: content_id, locale: locale)
+
     if item
-      handle_existing(item, base_path, message.delivery_info.routing_key)
+      existing_payload_version = item.publishing_api_payload_version || 0
+
+      if payload_version > existing_payload_version
```

Comment:
It feels better to move this logic into the item and be more descriptive int he method name:

```ruby
  item.old_version?(new_payload)
```

Reply:
I disagree - I think the model shouldn't have any knowledge of publishing API message formats. And this becomes a single comparison operation if I write a migration.